### PR TITLE
Added fix for a bug @Eugene found with the poller

### PR DESCRIPTION
**Pull request by dannylui**
Cannot always assume that hosts will get into 'contacted' state before 'dark' state.

### CODE REVIEW

File: lib/ansible/runner/poller.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def poll(self):
                 else:
                     self.runner.callbacks.on_async_ok(host, res, self.runner.vars_cache[host]['ansible_job_id'])
         for (host, res) in results['dark'].iteritems():
-            self.hosts_to_retry[host] += 1
+            self.hosts_to_retry[host] = self.hosts_to_retry[host] + 1 if self.hosts_to_retry.has_key(host) else 1
```
Comment (cchristous) [inline lib/ansible/runner/poller.py]:
```python
host in self.hosts_to_retry
```
is the preferred way to do this in Python nowadays.

**Comment by emuzykin** [comment]
@dannylui - any way you can test this bug fix?

File: lib/ansible/runner/poller.py (hunk 1 above)
Comment (dannylui) [inline lib/ansible/runner/poller.py]: Ty, fixed.

**Comment by cchristous** [comment]
provided @dannylui has a high confidence in this fix, it might be easiest to just merge this to master and re-run the jobs that use it.

**Comment by emuzykin** [comment]
Why is this getting merged into a release branch and not master?

**Comment by ferrants** [comment]
It's not. it says release1.7.2 in the pr

**Comment by emuzykin** [comment]
Have we moved to a workflow of working with ansible releases instead of everything working from master?

**Comment by ferrants** [comment]
@emuzykin - this is on our fork, used in a couple of jobs. Everything else is on ansible's fork at release1.7.1.

The process for upgrading is getting a new version of ansible on jenkins-staging and its slaves and running tests there. @babineaum is working on a jenkins smoke test

**Comment by ferrants** [comment]
Also, there is no ansible master branch, so i may be confused by what you mean

**Comment by emuzykin** [comment]
ok, :shipit: and i'll retest

**Comment by dannylui** [comment]
@emuzykin I've rested this bug fix in a small skytap environment.  The scenario is very edge case, the disconnect must happen after the first initial connection is established, but not the first poller update.

**Before**
```
[root@host-1 ~]# sleep 2; ansible-playbook -vvvv -i hosts.txt playbook.yml 

PLAY [all] ******************************************************************** 

GATHERING FACTS *************************************************************** 
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
<10.0.0.2> REMOTE_MODULE setup
<10.0.0.2> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1421950659.17-17970411919115 && echo $HOME/.ansible/tmp/ansible-tmp-1421950659.17-17970411919115'
<10.0.0.2> PUT /tmp/tmpeTu10W TO /root/.ansible/tmp/ansible-tmp-1421950659.17-17970411919115/setup
<10.0.0.2> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1421950659.17-17970411919115/setup; rm -rf /root/.ansible/tmp/ansible-tmp-1421950659.17-17970411919115/ >/dev/null 2>&1'
ok: [10.0.0.2]

TASK: [execute the tests] ***************************************************** 
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
<10.0.0.2> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511 && echo $HOME/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511'
<10.0.0.2> PUT /tmp/tmp6c3mGV TO /root/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511/command
<10.0.0.2> EXEC /bin/sh -c 'chmod a+rx /root/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511/command'
<10.0.0.2> PUT /tmp/tmpqnNzB5 TO /root/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511/async_wrapper
<10.0.0.2> PUT /tmp/tmpNE_dkJ TO /root/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511/arguments
<10.0.0.2> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511/async_wrapper 88483728825 120 /root/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511/command /root/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511/arguments; rm -rf /root/.ansible/tmp/ansible-tmp-1421950659.9-219177211336511/ >/dev/null 2>&1'
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
fatal: [10.0.0.2] => {'msg': 'FAILED: timed out', 'failed': True}
Traceback (most recent call last):
  File "/root/ansible/bin/ansible-playbook", line 309, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/root/ansible/bin/ansible-playbook", line 249, in main
    pb.run()
  File "/root/ansible/lib/ansible/playbook/__init__.py", line 339, in run
    if not self._run_play(play):
  File "/root/ansible/lib/ansible/playbook/__init__.py", line 695, in _run_play
    if not self._run_task(play, task, False):
  File "/root/ansible/lib/ansible/playbook/__init__.py", line 471, in _run_task
    results = self._run_task_internal(task)
  File "/root/ansible/lib/ansible/playbook/__init__.py", line 432, in _run_task_internal
    results = self._async_poll(poller, task.async_seconds, task.async_max_conseq_retries, task.async_poll_interval)
  File "/root/ansible/lib/ansible/playbook/__init__.py", line 356, in _async_poll
    results = poller.wait(async_seconds, async_max_conseq_retries, async_poll_interval)
  File "/root/ansible/lib/ansible/runner/poller.py", line 118, in wait
    poll_results = self.poll()
  File "/root/ansible/lib/ansible/runner/poller.py", line 89, in poll
    self.hosts_to_retry[host] += 1
KeyError: '10.0.0.2'
```

**After**
```
[root@host-1 ~]# sleep 2; ansible-playbook -vvvv -i hosts.txt playbook.yml 

PLAY [all] ******************************************************************** 

GATHERING FACTS *************************************************************** 
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
<10.0.0.2> REMOTE_MODULE setup
<10.0.0.2> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1421950819.96-81086733085186 && echo $HOME/.ansible/tmp/ansible-tmp-1421950819.96-81086733085186'
<10.0.0.2> PUT /tmp/tmpks20jV TO /root/.ansible/tmp/ansible-tmp-1421950819.96-81086733085186/setup
<10.0.0.2> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1421950819.96-81086733085186/setup; rm -rf /root/.ansible/tmp/ansible-tmp-1421950819.96-81086733085186/ >/dev/null 2>&1'
ok: [10.0.0.2]

TASK: [execute the tests] ***************************************************** 
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
<10.0.0.2> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509 && echo $HOME/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509'
<10.0.0.2> PUT /tmp/tmpGN_VVd TO /root/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509/command
<10.0.0.2> EXEC /bin/sh -c 'chmod a+rx /root/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509/command'
<10.0.0.2> PUT /tmp/tmpWhF1bR TO /root/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509/async_wrapper
<10.0.0.2> PUT /tmp/tmpPdjTwj TO /root/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509/arguments
<10.0.0.2> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509/async_wrapper 836046818584 120 /root/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509/command /root/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509/arguments; rm -rf /root/.ansible/tmp/ansible-tmp-1421950820.69-185069559026509/ >/dev/null 2>&1'
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
fatal: [10.0.0.2] => {'msg': 'FAILED: timed out', 'failed': True}
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
fatal: [10.0.0.2] => {'msg': 'FAILED: timed out', 'failed': True}
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
<10.0.0.2> REMOTE_MODULE async_status jid=836046818584.19506
<10.0.0.2> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1421950857.4-202591983318307 && echo $HOME/.ansible/tmp/ansible-tmp-1421950857.4-202591983318307'
<10.0.0.2> PUT /tmp/tmpZRdyFt TO /root/.ansible/tmp/ansible-tmp-1421950857.4-202591983318307/async_status
<10.0.0.2> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1421950857.4-202591983318307/async_status; rm -rf /root/.ansible/tmp/ansible-tmp-1421950857.4-202591983318307/ >/dev/null 2>&1'
<job 836046818584.19506> polling, 105s remaining
<10.0.0.2> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO 10.0.0.2
<10.0.0.2> REMOTE_MODULE async_status jid=836046818584.19506
<10.0.0.2> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1421950862.9-225310222035505 && echo $HOME/.ansible/tmp/ansible-tmp-1421950862.9-225310222035505'
<10.0.0.2> PUT /tmp/tmpE51qMc TO /root/.ansible/tmp/ansible-tmp-1421950862.9-225310222035505/async_status
<10.0.0.2> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1421950862.9-225310222035505/async_status; rm -rf /root/.ansible/tmp/ansible-tmp-1421950862.9-225310222035505/ >/dev/null 2>&1'
<job 836046818584.19506> polling, 100s remaining
...
```

**Pull request merged**